src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -2614,9 +2614,30 @@ public void onFailure(Throwable caught) {
   }
 
   private void onRename() {
-    checkUpdateEntity(() -> {
-      postCheckRename();
-    });
+    if (checkIsLatestVersion()) {
+      checkUpdateEntity(() -> {
+        postCheckRename();
+      });
+    } else {
+      view.showErrorMessage(
+        "Can only change the name of the most recent " +
+        entityBundle.getEntityType() +
+        " version."
+      );
```

Comment:
Now that I think about it, this will be confusing for datasets, because our terminology is different for those entities.

This text is fine for non-datasets, but if the entity is an instanceof Dataset, can you show an error message that says `"Can only change the name of the draft dataset"` ?